tools/rebuild_provision_request.rb

```diff
@@ -0,0 +1,311 @@
+require 'trollop'
+#
+# Helper Script to show the json / hash output of an
+# existing Automate request
+#
+#
+# Run without options to see the usage.
+#
+#
+
+PROGRAM_STRING = "rails runner #{$PROGRAM_NAME}".freeze
+
+unless Object.const_defined?(:Rails)
+  print "\nScript must be run with bin/rails runner\n"
+  print "\ne.g. #{PROGRAM_STRING} -- --help\n\n"
+  exit
+end
+
+if ARGV.empty?
+  print "\n#{PROGRAM_STRING} -- --help\n\n"
+  exit
+end
+
+ARGV.shift if ARGV[0] == '--'
+
+opts = Trollop.options do
+  banner <<-EOS
+
+Reconstruct provision request parameters based on an existing request
+
+Common Examples\n\t
+Output the hash parameters of a request
+
+  #{PROGRAM_STRING} --request-id=3
+
+Show the hash parameters and the console command needed to rerun a request
+
+  #{PROGRAM_STRING} --request-id=3  --console
+
+Show the hash parameters and rerun the request
+
+  #{PROGRAM_STRING} --request-id=3  --run-it
+
+Show a list of 5 recent requests
+
+  #{PROGRAM_STRING} --last-requests
+
+Help! #{PROGRAM_STRING} -- --help
+
+Usage: #{PROGRAM_STRING} [--options]\n\nOptions:\n\t
+
+  EOS
+  opt :api_host, "The hostname to run the api call against", :default => 'localhost', :type => :string
+  opt :console,  "Show the output needed to run the POST in a Rails console", :default => false, :type => :bool
+  opt :count, "Adjust the number of request ids",  :type => :int, :default => 5
```

Comment:
I thought `:count` only applied to the listing of requests.  If that is not true then it would be helpful to specify what other options it does have an effect on.